### PR TITLE
Add net6.0 as a target framework

### PR DESCRIPTION
**Pull request by Cristhian-Marin-asdk**
### Purpose

Add the dotnet 6.0 as a new target framework for the Greg Client nuget package

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers
@mjkkirschner 
@jasonstratton

### CODE REVIEW

File: src/AssemblyInfoGenerator/AssemblyInfo.cs [hunk 1]
```diff
@@ -1,58 +1,58 @@
...
+
+// Note that the assembly version does not get incremented for every build
+// to avoid problems with assembly binding (or requiring a policy or
+// <bindingRedirect> in the config file).
+//
+// The AssemblyFileVersionAttribute is incremented with every build in order
+// to distinguish one build from another. AssemblyFileVersion is specified
+// in AssemblyVersionInfo.cs so that it can be easily incremented by the
+// automated build process.
+[assembly: AssemblyVersion("2.4.0.5007")]
```
Comment (mjkkirschner) [inline src/AssemblyInfoGenerator/AssemblyInfo.cs:39]: so I would suggest actually updating the .tt file https://github.com/DynamoDS/PackageManagerClient/blob/master/src/AssemblyInfoGenerator/AssemblyInfo.tt#L62
to 2.5.0

File: src/GregClient/GregClient.csproj [hunk 2]
```diff
@@ -48,7 +48,7 @@
     <Reference Include="System.IO.Compression.FileSystem" />
     <Reference Include="System.Web" />
   </ItemGroup>
-  <ItemGroup Condition=" '$(TargetFramework)' == 'net5.0'">
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net5.0' Or '$(TargetFramework)' == 'net6.0'">
```
[the remark this answers is not in the document]
Comment (Cristhian-Marin-asdk) [reply src/GregClient/GregClient.csproj:51]: I tried to compile the nuget package with the same newtonsoft version as the .netframework, and seems to be working fine, also I tried to use the nuget package with DynamoCli, which seems to be working fine too, I didn't want to break something else, so I left the old newtonsoft version

File: src/AssemblyInfoGenerator/AssemblyInfo.cs (hunk 1 above)
Comment (Cristhian-Marin-asdk) [reply src/AssemblyInfoGenerator/AssemblyInfo.cs]: I have updated the file

File: src/GregClient/GregClient.csproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (mjkkirschner) [reply src/GregClient/GregClient.csproj:51]: I'd go for changing it to the same version - if it's an issue in the future it's easy to revert that - what do you think?

[the remark this answers is not in the document]
Comment (Cristhian-Marin-asdk) [reply src/GregClient/GregClient.csproj:51]: Sure, sounds fine to me, I'll do the change

File: src/GregClient/GregClient.csproj [hunk 3]
```diff
@@ -48,8 +48,8 @@
     <Reference Include="System.IO.Compression.FileSystem" />
     <Reference Include="System.Web" />
   </ItemGroup>
-  <ItemGroup Condition=" '$(TargetFramework)' == 'net5.0'">
-    <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net5.0' Or '$(TargetFramework)' == 'net6.0'">
```
[the remark this answers is not in the document]
Comment (Cristhian-Marin-asdk) [reply src/GregClient/GregClient.csproj:51]: Yes, we already have a common dependencies group, makes sense

**Review by mjkkirschner — APPROVED** [review]

**Comment by mjkkirschner** [comment]
@Cristhian-Marin-asdk please wait for the build to post a new Greg nuget to nuget.org and then update Dynamo's references to that version and send a PR to master - thanks!

**Pull request merged**